Whiskey/Functions/Invoke-WhiskeyTask.ps1

```diff
@@ -143,7 +143,64 @@ function Invoke-WhiskeyTask
             }
         }
     }
+    
+    $branch = $TaskContext.BuildMetadata.ScmBranch
+    $executeTaskOnBranch = $true
+    $onlyOnBranch = $null
+    $exceptOnBranch = $null
+    
+    if( $Parameter['OnlyOnBranch'] )
+    {
+        $executeTaskOnBranch =$false
+        $onlyOnBranch = $false
+        Write-Verbose -Message ('OnlyOnBranch')
+        foreach( $wildcard in $Parameter['OnlyOnBranch'] )
+        {
+            if( $branch -like $wildcard )
+            {
+                Write-Verbose -Message ('               {0}     -like  {1}' -f $branch, $wildcard)
+                $executeTaskOnBranch = $true
+                $onlyOnBranch = $true
+                break
+            }
+            else
+            {
+                Write-Verbose -Message ('               {0}  -notlike  {1}' -f $branch, $wildcard)
+            }
+        }
+    }
+
+    if( $Parameter['ExceptOnBranch'] )
+    {
+        $exceptOnBranch = $false
+        Write-Verbose -Message ('ExceptOnBranch')
+        foreach( $wildcard in $Parameter['ExceptOnBranch'] )
+        {
+            if( $branch -like $wildcard )
+            {
+                Write-Verbose -Message ('               {0}     -like  {1}' -f $branch, $wildcard)
+                $executeTaskOnBranch = $false
+                $exceptOnBranch = $true
+                break
+            }
+            else
+            {
+                Write-Verbose -Message ('               {0}  -notlike  {1}' -f $branch, $wildcard)
+            }
+        }
+    }
 
+    if( $onlyOnBranch -and $exceptOnBranch )

```

Comment:
Move this before you evaluate both parameters. Check `if( $Parameter['OnlyOnBranch'] -and $Parameter['ExceptOnBranch'] )`. Change error message to: `This task defines both OnlyOnBranch and ExceptOnBranch properties. Only one of these can be used. Please remove one or both of these properties and re-run your build.`